spyder/plugins/editor/widgets/tests/test_codeeditor.py

```diff
@@ -56,3 +57,57 @@ def test_editor_lower_to_upper(editorbot):
     widget.transform_to_uppercase()
     new_text = widget.get_text('sof', 'eof')
     assert text != new_text
+
+
+@pytest.mark.parametrize("input_text, expected_text, keypress",
+                         [('for i in range(2):\n    ',
+                           'for i in range(2):\n\n    ',
+                           Qt.Key_Enter),
+                          ('myvar = 2 ',
+                           'myvar = 2\n',
+                           Qt.Key_Enter),
+                          ('somecode = 1\nmyvar = 2 ',
+                           'somecode = 1\nmyvar = 2',
+                           Qt.Key_Up),
+                          ('somecode = 1\nmyvar = 2 ',
+                           'somecode = 1\nmyvar = 2 ',
+                           Qt.Key_Left),
+                          ('"""This is a string with important spaces\n    ',
+                           '"""This is a string with important spaces\n    \n',
+                           Qt.Key_Enter)
+                          ])
+def test_editor_rstrip_keypress(
+        editorbot, input_text, expected_text, keypress):
+    """Remove whitespace if we left a line with trailing whitespace
+    by keypress,"""
```

Comment:
This function tests that the whitespace is removed, not removes the whitespace itself; docstrings should end in a period (I made a small typo in my comment) and multi-line docstrings should have the ``"""``s on their own lines per our general convention, although this is *already* short enough to fit in one.

Therefore, I suggest:

``"""Test that whitespace is removed when leaving a line with it by keypress."""``